tauri-app/src/lib/stores/settings.ts

```diff
@@ -4,59 +4,97 @@ import  find from 'lodash/find';
 import * as chains from 'viem/chains';
 import { textFileStore } from '$lib/storesGeneric/textFileStore';
 import { invoke } from '@tauri-apps/api';
-import { type AppSettings, type ChainSettings } from '$lib/typeshare/appSettings';
-import { getBlockNumberFromRpc, getChainIdFromRpc } from '$lib/services/chain';
+import { type Config } from '$lib/typeshare/config';
+import { getBlockNumberFromRpc } from '$lib/services/chain';
 
-interface ChainSettingsExtended extends ChainSettings {
-  chain_id: number,
-}
+const emptyConfig = {
+  deployments: {},
+  networks: {},
+  orderbooks: {},
+  orders: {},
+  subgraphs: {},
+  tokens: {},
+  deployers: {},
+  scenarios: {},
+  charts: {}
+};
 
-interface AppSettingsExtended {
-  chains: Array<ChainSettingsExtended>;
-}
+// dotrain text store
+export const dotrainFile = textFileStore('Rain', ['rain']);
 
 // general
 export const settingsText = cachedWritableStore<string>('settings', "", (s) => s, (s) => s);
 export const settingsFile = textFileStore('Orderbook Settings Yaml', ['yml', 'yaml'], get(settingsText));
-export const settings = asyncDerived(settingsText, async ($settingsText): Promise<AppSettingsExtended> => {
-  const data: AppSettings = await invoke("parse_settings", {text: $settingsText});
-  const chains = await Promise.all(data.chains.map(async (c): Promise<ChainSettingsExtended> => {
-    const chainId: number = await getChainIdFromRpc(c.rpc_url);
-    return {
-      ...c,
-      chain_id: chainId,
-    };
-  }));
-  return { chains };
-});
+export const settings = asyncDerived([settingsText, dotrainFile], async ([$settingsText, $dotrainFile]): Promise<Config> => {
+  const text = $dotrainFile?.text !== undefined ? $dotrainFile.text : "";
+  try {
+    const config: Config = await invoke("get_config", {dotrain: text, settingText: $settingsText});
+    return config;
+  } catch(e) {
+    // eslint-disable-next-line no-console
+    console.log(e);
+    return emptyConfig;
+  }
+}, { initial: emptyConfig });
 
-// chain
-export const activeChainSettingsIndex = cachedWritableInt("settings.activeChainIndex", 0);
-export const activeChainSettings = derived([settings, activeChainSettingsIndex], ([$settingsData, $activeChainSettingsIndex]) => $settingsData?.chains[$activeChainSettingsIndex]);
-export const rpcUrl = derived(activeChainSettings, ($activeChainSettings) => $activeChainSettings?.rpc_url);
-export const chainId = derived(activeChainSettings, ($activeChainSettings) => $activeChainSettings?.chain_id);
+// networks
+export const networks = derived(settings, ($settingsData) => Object.entries($settingsData?.networks));
+export const activeNetworkIndex = cachedWritableInt("settings.activeNetworkIndex", 0);
```

Comment:
Now that networks is a `Record<string, Network>` I think it makes more sense to use the string key to identify the active network, rather than an index number.

Reply:
sure, I just tried to make the least amount of changes, so I dont break things unintentionally